rdt/transformers/utils.py

```diff
@@ -383,3 +385,39 @@ def _handle_enforce_uniqueness_and_cardinality_rule(enforce_uniqueness, cardinal
             result = 'unique'
 
     return result
+
+
+def _safe_parse_datetime(value):
+    if isinstance(value, (pd.Timestamp, datetime.datetime)):
+        return value
+    try:
+        return parser.parse(value)
+    except Exception:
```

Comment:
I don't see any [other blanket Exception catches](https://github.com/search?q=repo%3Asdv-dev%2FRDT+except+Exception&type=code) in RDT. Should we be more specific?